extensions/calDotCom/actions/__tests__/getBooking.ts

```diff
@@ -1,41 +1,188 @@
 
-  test('Should call the onComplete callback', async () => {
-    await getBooking.onActivityCreated(
-      {
-        pathway: {
-          id: 'pathway-id',
-          definition_id: 'pathway-definition-id',
+  describe('with empty bookingId', () => {
+    it('should call onError', async () => {
+      await getBooking.onActivityCreated(
+        {
+          ...dummyPayloadPart,
+          fields: {
+            bookingId: '',
+          },
+          settings: { apiKey: faker.string.uuid() },
+        },
+        onComplete,
+        onError
+      )
+      expect(onError).toHaveBeenCalledWith({
+        events: [
+          {
+            date: expect.any(String),
+            text: {
+              en: expect.stringContaining('Missing bookingId'),
+            },
+            error: {
+              category: 'BAD_REQUEST',
+              message: expect.stringContaining('Missing bookingId'),
+            },
+          },
+        ],
+      })
+    })
+  })
+
+  describe('with valid response', () => {
+    let mockCalComApi: jest.SpyInstance
+    let eventTypeId: number
+    let title: string
+    let description: string
+    let startTime: string
+    let endTime: string
+    let status: string
+    let uid: string
+
+    beforeEach(() => {
+      eventTypeId = faker.number.int()
+      title = faker.word.sample()
+      description = faker.word.sample()
+      startTime = faker.date.anytime().toISOString()
+      endTime = faker.date.anytime().toISOString()
+      status = faker.string.sample()
+      uid = faker.string.uuid()
+      mockCalComApi = jest
+        .spyOn(CalComApi.prototype, 'getBooking')
+        .mockResolvedValue({
+          eventTypeId,
+          title,
+          description,
+          startTime,
+          endTime,
+          status,
+          uid,
+        })
+    })
+
+    afterEach(() => {
+      mockCalComApi.mockRestore()
+    })
+
+    it('should call onComplete with data points', async () => {
+      await getBooking.onActivityCreated(
+        {
+          ...dummyPayloadPart,
+          fields: {
+            bookingId: faker.string.uuid(),
+          },
+          settings: { apiKey: faker.string.uuid() },
         },
-        activity: {
-          id: 'activity-id',
+        onComplete,
+        onError
+      )
+      expect(onComplete).toHaveBeenCalledWith({
+        data_points: {
+          eventTypeId: eventTypeId.toString(),
+          title,
+          description,
+          startTime,
+          endTime,
+          status,
+          cancelUrl: `https://app.cal.com/booking/${uid}?cancel=true`,
+          rescheduleUrl: `https://app.cal.com/reschedule/${uid}`,
         },
-        patient: { id: 'test-patient' },
-        fields: {
-          bookingId: '123',
+      })
+    })
+  })
+
+  describe('with error thrown from CalComApi.getBooking', () => {
+    let mockCalComApi: jest.SpyInstance
+    let errorMessage: string
+
+    beforeEach(() => {
+      errorMessage = faker.word.words()
```

Comment:
Just make this a static errorMessage

Reply:
I like faker better then coming up with strings and hardcoding them